### DISCUSSION TITLE
About the tropospheric delay correction

### DISCUSSION THREAD

**Discussion by ThreeIcug** [insarlab/MintPy#803] [General]
**Description of the desired feature**
 The Mintpy is a nice code for SBAS. Recently,  I try to process the Tibet data, none of the three methods of tropospheric delay correction seem to achieve a good result. So I wonder if the Global Atmospheric Models is not enough for plateau applications. Then, I found an paper of the Tymofyeyeva, E. and Y. Fialko (2015), they think the atmospheric noise is random, and estimate radar phase delays due to propagation through the troposphere and the ionosphere based on the averaging of redundant interferograms that share a common scene.  Are we considering adding a similar estimation method. 
My requirements may be a bit nitpicky.  

**Is your feature request related to a problem? Please describe**

**Describe the solution you'd like**

**Describe alternatives you have considered**

**Additional context**

**Are you willing to help implement and maintain this feature?** 

- [ ] Yes
- [ ] No

**Comment by mosyhey** [comment]
Dear @ThreeIcug,
If you have tested some atmospheric correction methods, would you please compare them briefly?
In your study area, which was the best? Did the GACOS method work better or the ERA5 method?

**Comment by scottstanie** [comment]
If you'd like to start playing around with how well it works, I have a repo that does the averaging to get a NetCDF stack of "average interferograms" (which are the rough estimates of the atmosphere on each SAR date): 
https://github.com/scottstanie/trodi

a few caveats:
- I haven't (yet) tried to integrate this to the MintPy or ISCE directory struction, but it's on my eventual TODOs
- similarly, it needs to have a way to incorporate nodata masks for interferograms (i've been doing that manually)

**Comment by ThreeIcug** [comment]
@mosyhey 
My study area is in the Qilian Mountains. I have tried three methods supported by Mintpy. I think the GACOS is the best in that area. Because the time series is smoother in space and time than other. But there are still problems in some areas. 
@scottstanie very kind of you, i will try it.

**Comment by yunjunz** [comment]
@ThreeIcug adding the method from Tymofyeyeva & Fialko (2015) is not on my to-do list, unfortunately, but contributions are welcomed. 

The power-law method from Bekaert et al. (2015, JGR) is another potential option and is available through [TRAIN](https://github.com/dbekaert/TRAIN). It's not directly compatible with mintpy format, but should be straightforward to make the connection.

The method from Cao et al. (2019, JGR) is another easy-to-use option, it's available here: https://github.com/ymcmrs/ICAMS and it's compatible with mintpy.

**Comment by mosyhey** [comment]
Evaluation of MintPy tropospheric delay correction methods by comparing InSAR info with CGPS data, in a case study area:

![los_insar_gps](https://user-images.githubusercontent.com/105506772/174427126-c6ee9c15-6f59-43d5-8836-e7b23dd18648.png)
